src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.ProjectPropertyStorage.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using EnvDTE;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem
+{
+    internal partial class VisualStudioWorkspaceImpl
+    {
+        private abstract class ProjectPropertyStorage
+        {
+            public static ProjectPropertyStorage Get(Project project, IServiceProvider serviceProvider)
```

Comment:
Maybe call this `Create` rather than `Get` to be clear that it always allocates?

Reply:
sure

Reply:
fixed in 28f3ee0d27a88bf412b16837832ab626baec3108